elastic/beats
# ci: speed builds for module specific changes

## What does this PR do?

Use the step to gather what's the module that has been changed when the changes are related to a specific module only. This will help to skip certain tests that are not related to the module that has been changes. 

## Why is it important?

Fast builds with the tests that required to be tested rather than everything else.

## How to test this PR locally

In the CI, touch certain files for the same module and see whether it does what's expected or the other way around, touch a few files and see if it works.

https://github.com/elastic/beats/pull/18592/commits/6b65f0541485d645b93744f8106699d4f752f906 caused 

```
[2020-05-16T09:39:25.049Z] [INFO] getBeatsModule: apache with regex [a-z0-9]+beat\/module\/([^\/]+)\/.*
```

See the very last line in the [log](https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats-beats-mbp/branches/PR-18592/runs/6/nodes/27/log/?start=0)
